table.go

```diff
@@ -78,28 +83,54 @@ func (rt *RoutingTable) GenRandPeerID(bucketID int) (peer.ID, error) {
 		targetCpl = bucketID
 	}
 
-	// should give up after a fixed number of attempts so we don't take up too much time/cpu
-	for i := 0; i < 1000; i++ {
-		peerID, err := randPeerID()
-		if err != nil {
-			log.Debugf("failed to generate random peerID in bucket %d, error is %+v", bucketID, err)
-			continue
+	// generate random 16 bits
+	r := rand.New(rand.NewSource(time.Now().UnixNano()))
```

Comment:
Just single call to rand after `rand.New` can be biased. The `rand` should probably be in `*RoutingTable`.

Reply:
@Kubuxu  That sounds interesting. Even if I seed it with the current time in nanos ?

Reply:
Yes, pseudo-random algorithms provide pseudo-random numbers if you use them for some time. There can be an initial bias if you use only the one number from the prng. It depends on the implementation of the prng itself.

Reply:
Unless we want to be able to manipulate the random number (i.e. for testing) why don't we just use rand's built in global? Are we concerned about contention on that lock?

Reply:
Done